src/components/ItemSummary.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { ItemData } from "../LibraryUtilities";
+import { LibraryContainer } from "./LibraryContainer";
+
+interface ItemSummaryProps {
+    libraryContainer: LibraryContainer,
+    data: ItemData;
+    showDescription: boolean;
+}
+
+interface ItemSummaryStates {
+    hasItemSummaryData: boolean;
+}
+
+export class ItemSummary extends React.Component<ItemSummaryProps, ItemSummaryStates> {
+    itemSummaryData: any;
+
+    constructor(props: ItemSummaryProps) {
+        super(props);
+        this.state = ({ hasItemSummaryData: false });
+        this.itemSummaryData = null;
```

Comment:
A very minor thing, since we are already in the class `ItemSummary`, the names of data members can be simplified. For example, `hasSummaryData` and `this.summaryData`.